_studio/mfx_lib/encode_hw/h265/include/mfx_h265_encode_hw_utils.h

```diff
@@ -77,7 +77,7 @@ template<class T> inline void Zero(T * first, size_t cnt)     { memset(first, 0,
 template<class T, class U> inline void Copy(T & dst, U const & src)
 {
     static_assert(sizeof(T) == sizeof(U), "copy_objects_of_different_size");
-    memcpy_s(&dst, sizeof(dst), &src, sizeof(dst));
+    std::copy(std::begin(src), std::end(src), std::begin(dst));
```

Comment:
no, I disagree. Passed arguments might have any type. It could even be non-POD, so previous code was super dangerous. Remove this template function at all and fix copying in place everywhere it used. I don't see other way

Reply:
I wonder, do we have the same function declared in AVC and VP9e utils?

Reply:
This file has more dangerous templates, like Zero and Fill. + many other functions which are candidates for clean up (max/min/abs, even sort on macro)

I agree that we need to check other files

Reply:
Yep. I remember that there is an awful duplication of utility code between our encoders. That's really a chance to clean up this mess. Need to plan the activity though. No need to mix everything together. If the idea is to remove self-written wrappers in a preference of std functions - this will be a good step forward by itself.